ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/PullQueryMetricsTest.java

```diff
@@ -313,14 +326,16 @@ private double getMetricValue(
       final String metricName,
       final PullSourceType sourceType,
       final PullPhysicalPlanType planType,
-      final RoutingNodeType routingNodeType
+      final RoutingNodeType routingNodeType,
+      final DataSourceType dataSourceType
   ) {
     final Metrics metrics = pullMetrics.getMetrics();
     final Map<String, String> tags = ImmutableMap.<String, String>builder()
         .putAll(CUSTOM_TAGS_WITH_SERVICE_ID)
         .put(KsqlConstants.KSQL_QUERY_SOURCE_TAG, sourceType.name().toLowerCase())
         .put(KsqlConstants.KSQL_QUERY_PLAN_TYPE_TAG, planType.name().toLowerCase())
         .put(KsqlConstants.KSQL_QUERY_ROUTING_TYPE_TAG, routingNodeType.name().toLowerCase())
+        .put(KsqlConstants.KSQL_DATA_SOURCE_TYPE_TAG, dataSourceType.name().toLowerCase())
```

Comment:
The fact that we have to modify the expectations of this test is making me nervous that this might not be a compatible change.

It would be safer, though more verbose, to just add a whole new set of identical metrics for stream pull query metrics.

Reply:
I'm not sure if this is compatible.  I would think it should be but might be worth checking.  I think this is preferable to creating a whole new set of metrics.